src/frontend/src/handler/create_source.rs

```diff
@@ -113,20 +119,43 @@ pub async fn handle_create_source(
 
     let mut with_properties = context.with_options.inner().clone();
 
+    // pass `use_schema_registry` to `SourceParserImpl::create` by insert it into `with_properties`
+    const USE_SCHEMA_REGISTRY: &str = "use_schema_registry";
```

Comment:
downvote for this. suggest to add a field in `StreamSourceInfo` (proto/catalog.proto)

Reply:
we should be cautious to pass info in with_properties